### PR TITLE
EVG-19925: Upgrade to go 1.20, and update distros.

### PR DESCRIPTION
**Pull request by johndaniels**

### CODE REVIEW

File: evergreen.yaml [hunk 1]
```diff
@@ -168,31 +168,20 @@ post:
...
-      RACE_DETECTOR: true
-    run_on:
-      - archlinux-new-small
-      - archlinux-new-large
-    tasks:
-      - ".test"
 
   - name: lint
-    display_name: Lint (Arch Linux)
+    display_name: Lint (Ubuntu 22.04)
```
Comment (Kimchelly) [inline evergreen.yaml:173]: We can just call this `Lint`, since the particular Linux distro isn't that relevant.

**Review by Kimchelly — APPROVED** [review]
LGTM mod small comments.

File: doc.go [hunk 2]
```diff
@@ -2,12 +2,12 @@
 Package curator is a tool for generating and publishing operating
 system packages (e.g. deb and rpm) to S3 buckets.
 
-Architecture and Organization
+# Architecture and Organization
```
[the remark this answers is not in the document]
Comment (johndaniels) [reply doc.go:5]: Where do the generated godocs go? https://go.dev/doc/comment indicates that it handles headings, and I don't know why gofmt would put them in if they didn't work?

[the remark this answers is not in the document]
Comment (Kimchelly) [reply doc.go:5]: The godocs are in `https://pkg.go.dev/<module_name>` ([this](https://pkg.go.dev/github.com/mongodb/curator)). I guess if gofmt wants it this way, then it's fine.

[the remark this answers is not in the document]
Comment (johndaniels) [reply doc.go:5]: Yeah I think that tool is supposed to recognize headers?

File: evergreen.yaml (hunk 1 above)
Comment (johndaniels) [reply evergreen.yaml]: Fixed

**Pull request merged**